### PR TITLE
nstat input plugin

### PR DESCRIPTION
**Pull request by Maksadbek**
Plugin discussed here: https://github.com/influxdata/telegraf/issues/1104

@sparrc Plugin for nstat, could not use your nstat library because there was a bit misfit and required refactoring, but I used the your parsers.

### CODE REVIEW

File: plugins/inputs/nstat/nstat.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+type Nstat struct {
+	ProcNetNetstat string `toml:"proc_net_netstat"`
+	ProcNetSNMP    string `toml:"proc_net_snmp"`
+	ProcNetSNMP6   string `toml:"proc_net_snmp6"`
+	DumpZeros      bool   `toml:"dump_zeros"`
+}
+
+var sampleConfig = `
+	# file paths
+	proc_net_netstat 	= 	"/proc/net/netstat"
```
Comment (sparrc) [inline plugins/inputs/nstat/nstat.go]: Here I think we should do the same thing nstat does: https://github.com/shemminger/iproute2/blob/master/misc/nstat.c#L46,

which is:

1. Use overrides here (but default to "")
2. try PROC_NET_NETSTAT env var
3. try PROC_ROOT env var + "net/netstat"
4. use "/proc/net/netstat"

File: plugins/inputs/nstat/nstat.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+type Nstat struct {
+	ProcNetNetstat string `toml:"proc_net_netstat"`
+	ProcNetSNMP    string `toml:"proc_net_snmp"`
+	ProcNetSNMP6   string `toml:"proc_net_snmp6"`
+	DumpZeros      bool   `toml:"dump_zeros"`
+}
+
+var sampleConfig = `
+	# file paths
```
Comment (sparrc) [inline plugins/inputs/nstat/nstat.go]: sampleConfig should be indented by two spaces

File: plugins/inputs/nstat/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## Nstat input plugin
```
Comment (sparrc) [inline plugins/inputs/nstat/README.md]: readme will need to be improved, 
model the readme after https://github.com/influxdata/telegraf/blob/master/plugins/inputs/EXAMPLE_README.md

**Comment by Maksadbek** [comment]
Hi @sparrc, I have fixed code, the paths for files are read as follows:
1. Try to read from config, default is ""
2. Try to read paths from env
3. Try to read PROC_ROOT and append default paths
4. use default paths: /proc/net/netstat, /proc/net/snmp, /proc/net/snmp6

Wrote detailed documentation, what fields are collected and their tags.
Feel free to comment if I missed smth, any your suggestions are appreciated.

**Pull request closed without merging**